app/org/sagebionetworks/bridge/dynamodb/DynamoScheduledActivityDao.java

```diff
@@ -131,5 +139,23 @@ public void deleteActivities(String healthCode) {
             BridgeUtils.ifFailuresThrowException(failures);
         }
     }
+    
+    /** {@inheritDoc} */
+    @Override
+    public void deleteActivitiesForSchedulePlan(String schedulePlanGuid) {
+        ScheduledActivity activity = new DynamoScheduledActivity();
+        activity.setSchedulePlanGuid(schedulePlanGuid);
+
+        List<ScheduledActivity> activitiesToDelete = schedulePlanIndex
+                .query(DynamoScheduledActivity.class, "schedulePlanGuid", schedulePlanGuid, null)
+                .stream()
+                .filter(act -> ScheduledActivityStatus.DELETABLE_STATUSES.contains(act.getStatus()))
```

Comment:
Can we add a unit test to test that activities without deletable statuses aren't deleted?